src/components/Notifications/Form/IntegrationRecipientTypeahead.tsx

```diff
@@ -53,7 +59,12 @@ export const IntegrationRecipientTypeahead: React.FunctionComponent<IntegrationR
         }
     }, [ getIntegrations, props.integrationType, state.loadingFilter, state.lastSearch, dispatchers ]);
 
-    const options = useRecipientOptionMemo(state);
+    const getExistingIntegrations = React.useCallback(() => {
```

Comment:
This could be an `useMemo`. That way we wouldn't need to call it on every render and the value would be stable.

Reply:
My understanding was that useCallback memoizes the return value while useMemo memoizes the function itself, would useCallback not be the correct one in this instance?

Reply:
[useCallback](https://reactjs.org/docs/hooks-reference.html#usecallback) will memoize the function itself - This is useful when we want to pass the function as an argument to other component or when we want to use the the function as a dependency.

[useMemo](https://reactjs.org/docs/hooks-reference.html#usememo) will memoize the result of the function itself.

As found in the docs:
> `useCallback(fn, deps) is equivalent to useMemo(() => fn, deps).`

Reply:
Ok, it seems that I had them backwards.